lib/cylc/network/httpclient.py

```diff
@@ -241,15 +249,76 @@ def put_ext_trigger(self, event_message, event_id):
 
-    def put_message(self, task_id, severity, message):
-        """Send task message."""
-        func_name = self._compat('put_message')
-        if func_name == 'put_message':
-            return self._call_server(
-                func_name, task_id=task_id, severity=severity, message=message)
-        else:  # pre-7.5.0 API compat
-            return self._call_server(
-                func_name, task_id=task_id, priority=severity, message=message)
+    def put_messages(self, payload):
+        """Send task messages to suite server program.
+
+        Arguments:
+            payload (dict):
+                task_job (str): Task job as "CYCLE/TASK_NAME/SUBMIT_NUM".
+                event_time (str): Event time as string.
+                messages (list): List in the form [[severity, message], ...].
+        """
+        retry_intvl = float(self.comms1.get(
+            self.srv_files_mgr.KEY_TASK_MSG_RETRY_INTVL,
+            self.MSG_RETRY_INTVL))
+        max_tries = int(self.comms1.get(
+            self.srv_files_mgr.KEY_TASK_MSG_MAX_TRIES,
+            self.MSG_MAX_TRIES))
+        for i in range(1, max_tries + 1):  # 1..max_tries inclusive
+            orig_timeout = self.timeout
+            if self.timeout is None:
+                self.timeout = self.MSG_TIMEOUT
+            try:
+                func_name = self._compat('put_messages')
+                if func_name == 'put_messages':
+                    results = self._call_server(func_name, payload=payload)
+                elif func_name == 'put_message':  # API 1, 7.5.0 compat
+                    cycle, name = payload['task_job'].split('/')[0:2]
+                    for severity, message in payload['messages']:
+                        results.append(self._call_server(
+                            func_name, task_id='%s.%s' % (name, cycle),
+                            severity=severity, message=message))
+                else:  # API 0, pre-7.5.0 compat, priority instead of severity
+                    cycle, name = payload['task_job'].split('/')[0:2]
+                    for severity, message in payload['messages']:
+                        results.append(self._call_server(
+                            func_name, task_id='%s.%s' % (name, cycle),
+                            priority=severity, message=message))
+            except ClientInfoError:
+                raise
+            except ClientError as exc:
+                now = get_current_time_string()
+                sys.stderr.write(
+                    "%s WARNING - Message send failed, try %s of %s: %s\n" % (
+                        now, i, max_tries, exc))
+                # Break if:
+                # * Exhausted number of tries.
+                # * Contact info file not found, suite probably not running.
+                #   Don't bother with retry, suite restart will poll any way.
+                if i >= max_tries or isinstance(exc, ClientInfoError):
```

Comment:
``ClientInfoError`` is raised separately prior to it's base exception class ``ClientError`` on 287:288, so I believe the second condition will never be ``True`` as it would have already have been caught?